src/com/trilead/ssh2/transport/TransportConnection.java

```diff
@@ -125,19 +125,21 @@ public void sendMessage(byte[] message, int off, int len, int padd) throws IOExc
 		else if (padd > 64)
 			padd = 64;
 
-		int packet_len = 5 + len + padd; /* Minimum allowed padding is 4 */
+		boolean encryptThenMac = send_mac != null && send_mac.isEncryptThenMac();
 
-		int slack = packet_len % send_padd_blocksize;
+		int encryptedPacketLength = (encryptThenMac ? 1 : 5) + len + padd; /* Minimum allowed padding is 4 */
```

Comment:
the name of the variable `encryptThenMac` is not accurate, it is not always encrypted. I'd keep the value as it was `packet_len`

Reply:
@kuisathaverat thx for comment. I will change back to using, packet_len instead of packet_len.